app/src/lib/repository-matching.ts

```diff
@@ -92,3 +94,22 @@ export function matchExistingRepository(
     }) || null
   )
 }
+
+/**
+ * Check a GitHub repository matches a given remote.
+ *
+ * The remote's URL might match the cloneURL field of the repository (with a
+ * .git suffix) or it might match the htmlURL field (without a .git suffix)
+ *
+ * @param gitHubRepository the GitHub repository details found from the GitHub API
+ * @param remote the remote details found in the Git repository
+ */
+export function repositoryMatchesRemote(
+  gitHubRepository: GitHubRepository,
+  remote: IRemote
+): boolean {
+  return (
+    gitHubRepository.cloneURL === remote.url ||
+    gitHubRepository.htmlURL === remote.url
```

Comment:
Should we make these checks case-insensitive given that they're GitHub specific? Also, should we approach this the same way we do in `matchRemoteWithAccount` i.e. not care about the protocol?

Reply:
:thumbsup: to case-insensitve

> i.e. not care about the protocol?

I think the only place where this will come is with `cloneURL` not matching `remote.url` when they are SSH/HTTPS. I'll give it a shot and see how it feels